File: frond.end/src/app/productos/producto.ts
import { Categoria } from "../categorias/categoria";

export class Producto {
  id:number;
  nombre:String;
  descripcion:String;
  precio:String;
  marca:String;
  modelo:String;
  categoriaId: number;
  categoria: Categoria; // Objeto de tipo Categoria

//   constructor(data?: any) {
//     if (data) {
//       this.id = data.id;
//       this.nombre = data.nombre;
//       this.descripcion = data.descripcion;
//       this.precio = data.precio;
//       this.marca = data.marca;
//       this.modelo = data.modelo;
//       this.categoria = new Categoria(data.categoria);
//   }
//  }
}

File: catalogo/src/main/java/eshop/catalogo/entity/Producto.java
package eshop.catalogo.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;
@Entity
@Data
public class Producto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String nombre;
    private String descripcion;
    private String precio;
    private String marca;
    private String modelo;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "categoria_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
    private Categoria categoria;
}